Pointwise.lua

```diff
@@ -31,11 +31,14 @@ function Pointwise:createIODescriptors(input)
    end
 
    local nElem = input:nElement()
-   self.nElem = self.nElem or nElem -- this goes to the second branch only once
-   if self.iDesc and nElem == self.nElem then return end
-   self.nElem = nElem
-   self.iDesc = cudnn.toDescriptor(input:view(1,1,1,nElem))
-
+   self.nElem = self.nElem or nElem
+   if self.iDesc and nElem == self.nElem and
```

Comment:
Input tensor can change between runs, so I think you can't omit creating a descriptor for it.